cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1355,57 +1356,7 @@ private StatusMap getStatus(final Id.Program id, final ProgramType type) {
                              HttpResponseStatus.NOT_FOUND.getCode());
       }
 
-      // For program type other than MapReduce
-      if (type != ProgramType.MAPREDUCE) {
-        return getProgramStatus(id, type, new StatusMap());
-      }
-
-      // must do it this way to allow anon function in workflow to modify status
-      // check that mapreduce exists
-      if (!appSpec.getMapReduce().containsKey(id.getId())) {
-        return new StatusMap(null, "Program: " + id.getId() + " not found", HttpResponseStatus.NOT_FOUND.getCode());
-      }
-
-      // See if the MapReduce is part of a workflow
-      String workflowName = getWorkflowName(id.getId());
-      if (workflowName == null) {
-        // Not from workflow, treat it as simple program status
-        return getProgramStatus(id, type, new StatusMap());
-      }
-
-      // MapReduce is part of a workflow. Query the status of the workflow instead
-      final SettableFuture<StatusMap> statusFuture = SettableFuture.create();
-      workflowClient.getWorkflowStatus(id.getNamespaceId(), id.getApplicationId(),
-                                       workflowName, new WorkflowClient.Callback() {
-          @Override
-          public void handle(WorkflowClient.Status status) {
-            StatusMap result = new StatusMap();
-
-            if (status.getCode().equals(WorkflowClient.Status.Code.OK)) {
-              result.setStatus("RUNNING");
-              result.setStatusCode(HttpResponseStatus.OK.getCode());
-            } else {
-              //mapreduce name might follow the same format even when its not part of the workflow.
-              try {
-                // getProgramStatus returns program status or http response status NOT_FOUND
-                getProgramStatus(id, type, result);
-              } catch (Exception e) {
-                LOG.error("Exception raised when getting program status for {} {}", id, type, e);
-                // error occurred so say internal server error
-                result.setStatusCode(HttpResponseStatus.INTERNAL_SERVER_ERROR.getCode());
-                result.setError(e.getMessage());
-              }
-            }
-
-            // This would make all changes in the result statusMap available to the other thread that doing
-            // the take() call.
-            statusFuture.set(result);
-          }
-        }
-      );
-      // wait for status to come back in case we are polling mapreduce status in workflow
-      // status map contains either a status or an error
-      return Futures.getUnchecked(statusFuture);
```

Comment:
Deleted this code since it assumes the MAPREDUCE program running inside WORKFLOW is named as WORKFLOW_MAPREDUCE and then returns the status of the WORKFLOW instead of the MAPREDUCE program. In 2.7 we moved MAPREDUCE to application level, so their names no longer contain the WORKFLOW name. This will be fixed with jira - https://issues.cask.co/browse/CDAP-1293